lib/index.ts

```diff
@@ -175,14 +175,14 @@ export class Checker {
     }
   }
 
-  private _doValidate(checkerFunc: CheckerFunc, value: any): IErrorDetail|null {
+  private _doValidate(checkerFunc: CheckerFunc, value: any): IErrorDetail[] {
     const noopCtx = new NoopContext();
     if (checkerFunc(value, noopCtx)) {
-      return null;
+      return [];
```

Comment:
My first instinct was that one should generally favour empty over null, but I realise now that doesn't apply well here. There should be a strong difference between the two cases. Especially since JS sees empty lists as truthy :angry: So I'll revert this. I think your reasoning is good and that this library should cater much more to typescript users than plain-JS.